File: frameowkrs/java-selenium-rest-assured/pages/CartPage.java
package pages;

import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.WebDriver;

public final class CartPage extends BasePage {
    public CartPage(WebDriver driver) { super(driver); }

    public void openAndVerify(String productName, int quantity, double unitPrice) {
        open("/cart");
        String items = visible("cart-items").getText();
        Assertions.assertTrue(items.contains(productName));
        Assertions.assertTrue(items.contains(String.valueOf(quantity)));
        Assertions.assertTrue(visible("cart-subtotal").getText()
                .contains("%.2f".formatted(quantity * unitPrice)));
    }

    public void proceedToCheckout() { visible("cart-checkout").click(); }
}

File: frameowkrs/typescript-cypress/pages/cart_page.ts
export class CartPage {
  openAndVerify(productName: string, quantity: number, unitPrice: number): void {
    cy.visit('/cart');
    cy.get('[data-testid="cart-items"]').should('contain.text', productName).and('contain.text', quantity);
    cy.get('[data-testid="cart-subtotal"]').should('contain.text', (quantity * unitPrice).toFixed(2));
  }

  proceedToCheckout(): void {
    cy.get('[data-testid="cart-checkout"]').click();
  }
}
